### ISSUE TITLE
Palemoon 27 unable to install - conflict with currently installed one?

### ISSUE THREAD

**Issue by hkmaly** [deu/palemoon-overlay#48]
After compiling palemoon, the install phase fails with error. I think I already had same problem with 27.1.0 and now with 27.3.0. The error message looks like something is looking at /usr/lib/palemoon/libxul.so which at that point is still from currently installed version of palemoon (www-client/palemoon-26.5.0).

Note: I'll now try to remove it and compile and install again, but I think it still should be fixed.

```
>>> Install palemoon-27.3.0 into /var/tmp/portage/www-client/palemoon-27.3.0/image/ category www-client
 * PT_PAX marking -m obj-i686-pc-linux-gnu/dist/bin/xpcshell with paxctl
 * PT_PAX marking -m obj-i686-pc-linux-gnu/dist/bin/xpcshell with scanelf
 * XATTR_PAX marking -me obj-i686-pc-linux-gnu/dist/bin/xpcshell with setfattr
 * Creating the package...
 0:00.71 /usr/bin/gmake -C . -j4 -s -w package
 0:00.86 gmake: Entering directory `/var/tmp/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0/obj-i686-pc-linux-gnu'
 0:00.87 gmake[1]: Entering directory `/var/tmp/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0/obj-i686-pc-linux-gnu/browser/installer'
 0:01.09 gmake[2]: Entering directory `/var/tmp/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0/obj-i686-pc-linux-gnu/browser/installer'
 0:01.19 gmake[3]: Entering directory `/var/tmp/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0/obj-i686-pc-linux-gnu/browser/installer'
 0:14.69 Executing /var/tmp/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0/obj-i686-pc-linux-gnu/dist/bin/xpcshell -g /var/tmp/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0/obj-i686-pc-linux-gnu/dist/bin/ -a /var/tmp/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0/obj-i686-pc-linux-gnu/dist/bin/ -f /var/tmp/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0/toolkit/mozapps/installer/precompile_cache.js -e precompile_startupcache("resource://gre/");
 0:14.69 /var/tmp/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0/obj-i686-pc-linux-gnu/dist/bin/xpcshell: /usr/lib/palemoon/libxul.so: version `xul3' not found (required by /var/tmp/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0/obj-i686-pc-linux-gnu/dist/bin/xpcshell)
 0:14.70 Traceback (most recent call last):
 0:14.70   File "/var/tmp/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0/toolkit/mozapps/installer/packager.py", line 405, in <module>
 0:14.70     main()
 0:14.70   File "/var/tmp/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0/toolkit/mozapps/installer/packager.py", line 399, in main
 0:14.70     args.source, gre_path, base)
 0:14.70   File "/var/tmp/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0/toolkit/mozapps/installer/packager.py", line 157, in precompile_cache
 0:14.70     errors.fatal('Error while running startup cache precompilation')
 0:14.70   File "/var/tmp/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0/python/mozbuild/mozpack/errors.py", line 101, in fatal
 0:14.70     self._handle(self.FATAL, msg)
 0:14.70   File "/var/tmp/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0/python/mozbuild/mozpack/errors.py", line 96, in _handle
 0:14.70     raise ErrorMessage(msg)
 0:14.70 mozpack.errors.ErrorMessage: Error: Error while running startup cache precompilation
 0:14.71 gmake[3]: *** [stage-package] Error 1
 0:14.71 gmake[3]: Leaving directory `/var/tmp/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0/obj-i686-pc-linux-gnu/browser/installer'
 0:14.71 gmake[2]: *** [make-package] Error 2
 0:14.71 gmake[2]: Leaving directory `/var/tmp/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0/obj-i686-pc-linux-gnu/browser/installer'
 0:14.71 gmake[1]: *** [default] Error 2
 0:14.71 gmake[1]: Leaving directory `/var/tmp/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0/obj-i686-pc-linux-gnu/browser/installer'
 0:14.71 gmake: *** [package] Error 2
 0:14.71 gmake: Leaving directory `/var/tmp/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0/obj-i686-pc-linux-gnu'
 * ERROR: www-client/palemoon-27.3.0::palemoon failed (install phase):
 *   (no error message)
 * 
 * Call stack:
 *     ebuild.sh, line 133:  Called src_install
 *   environment, line 3784:  Called die
 * The specific snippet of code:
 *       python2 mach package || die;
 * 
 * If you need support, post the output of `emerge --info '=www-client/palemoon-27.3.0::palemoon'`,
 * the complete build log and the output of `emerge -pqv '=www-client/palemoon-27.3.0::palemoon'`.
 * The complete build log is located at '/var/log/portage/www-client:palemoon-27.3.0:20170701-232718.log'.
 * For convenience, a symlink to the build log is located at '/var/tmp/portage/www-client/palemoon-27.3.0/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/www-client/palemoon-27.3.0/temp/environment'.
 * Working directory: '/var/tmp/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0'
 * S: '/var/tmp/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0'

>>> Failed to emerge www-client/palemoon-27.3.0, Log file:
```

**Comment by hkmaly** [comment]
```
# emerge --info '=www-client/palemoon-27.3.0::palemoon'
Portage 2.2.26 (python 2.7.12-final-0, default/linux/x86/13.0, gcc-4.9.3, glibc-2.23-r3, 3.13.0-100-generic i686)
=================================================================
                         System Settings
=================================================================
System uname: Linux-3.13.0-100-generic-i686-AMD_Phenom-tm-_II_X3_720_Processor-with-gentoo-2.2
KiB Mem:    12437600 total,   6692008 free
KiB Swap:          0 total,         0 free
Timestamp of repository gentoo: Sat, 01 Jul 2017 22:15:01 +0000
sh bash 4.3_p48-r1
ld GNU ld (GNU Binutils) 2.23.2
distcc 3.1 i686-pc-linux-gnu [disabled]
app-shells/bash:          4.3_p48-r1::gentoo
dev-java/java-config:     2.2.0-r3::gentoo
dev-lang/perl:            5.22.3_rc4::gentoo
dev-lang/python:          2.7.12::gentoo, 3.4.5::gentoo
dev-util/cmake:           3.7.2::gentoo
dev-util/pkgconfig:       0.28-r1::gentoo
sys-apps/baselayout:      2.2::gentoo
sys-apps/openrc:          0.11.8::gentoo
sys-apps/sandbox:         2.10-r1::gentoo
sys-devel/autoconf:       2.13::gentoo, 2.69::gentoo
sys-devel/automake:       1.4_p6-r1::gentoo, 1.6.3-r1::gentoo, 1.8.5-r4::gentoo, 1.9.6-r3::gentoo, 1.10.3::gentoo, 1.11.6::gentoo, 1.12.6::gentoo, 1.13.4::gentoo, 
1.14.1::gentoo, 1.15::gentoo
sys-devel/binutils:       2.18-r3::gentoo, 2.20.1-r1::gentoo, 2.23.2::gentoo, 2.25.1-r1::gentoo
sys-devel/gcc:            2.95.3-r8::<unknown repository>, 3.4.6-r2::gentoo, 4.5.3-r2::gentoo, 4.6.3::gentoo, 4.7.3-r1::gentoo, 4.8.5::gentoo, 4.9.3::gentoo
sys-devel/gcc-config:     1.7.3::gentoo
sys-devel/libtool:        2.4.6::gentoo
sys-devel/make:           3.82-r4::gentoo
sys-kernel/linux-headers: 3.13::gentoo (virtual/os-headers)
sys-libs/glibc:           2.23-r3::gentoo
Repositories:

gentoo
    location: /usr/portage
    sync-type: rsync
    sync-uri: rsync://rsync.europe.gentoo.org/gentoo-portage
    priority: -1000

sunrise
    location: /usr/local/portage/layman/sunrise
    masters: gentoo
    priority: 0

dev-zero
    location: /usr/local/portage/layman/dev-zero
    masters: gentoo
    priority: 1

palmer
    location: /usr/local/portage/layman/palmer
    masters: gentoo
    priority: 2

palemoon
    location: /usr/local/portage/layman/palemoon
    masters: gentoo
    priority: 3

x11
    location: /usr/local/portage/layman/x11
    masters: gentoo
    priority: 4

x-portage
    location: /usr/local/portage
    masters: gentoo
    priority: 5

ACCEPT_KEYWORDS="x86"
ACCEPT_LICENSE="* -@EULA"
CBUILD="i686-pc-linux-gnu"
CFLAGS="-O2 -mtune=k8 -march=k8 -fomit-frame-pointer -pipe"
CHOST="i686-pc-linux-gnu"
CONFIG_PROTECT="/etc /usr/lib/libreoffice/program/sofficerc /usr/share/easy-rsa /usr/share/gnupg/qualified.txt /var/bind /var/lib/hsqldb"
CONFIG_PROTECT_MASK="/etc/ca-certificates.conf /etc/dconf /etc/env.d /etc/fonts/fonts.conf /etc/gconf /etc/gentoo-release /etc/php/apache2-php5.3/ext-active/ /etc/php/apache2-php5.4/ext-active/ /etc/php/apache2-php5.6/ext-active/ /etc/php/cgi-php5.3/ext-active/ /etc/php/cgi-php5.4/ext-active/ /etc/php/cgi-php5.6/ext-active/ 
/etc/php/cli-php5.3/ext-active/ /etc/php/cli-php5.4/ext-active/ /etc/php/cli-php5.6/ext-active/ /etc/revdep-rebuild /etc/sandbox.d /etc/terminfo /etc/texmf/languag
e.dat.d /etc/texmf/language.def.d /etc/texmf/updmap.d /etc/texmf/web2c"
CXXFLAGS="-O2 -mtune=k8 -march=k8 -fomit-frame-pointer -pipe"
DISTDIR="/usr/portage/distfiles"
FCFLAGS="-O2 -march=i686 -pipe"
FEATURES="assume-digests binpkg-logs config-protect-if-modified distlocks ebuild-locks fixlafiles merge-sync news parallel-fetch preserve-libs protect-owned sandbo
x sfperms strict unknown-features-warn unmerge-logs unmerge-orphans userfetch userpriv usersandbox usersync xattr"
FFLAGS="-O2 -march=i686 -pipe"
GENTOO_MIRRORS="http://gentoo.mirror.dkm.cz/pub/gentoo/ ftp://ftp.fi.muni.cz/pub/linux/gentoo/ ftp://mirror.switch.ch/mirror/gentoo/ http://trumpetti.atm.tut.fi/ge
ntoo/ http://gentoo.oregonstate.edu http://www.ibiblio.org/pub/Linux/distributions/gentoo"
LDFLAGS="-Wl,-O1 -Wl,--as-needed"
MAKEOPTS="-j4"
PKGDIR="/usr/portage/packages"
PORTAGE_CONFIGROOT="/"
PORTAGE_RSYNC_OPTS="--recursive --links --safe-links --perms --times --omit-dir-times --compress --force --whole-file --delete --stats --human-readable --timeout=1
80 --exclude=/distfiles --exclude=/local --exclude=/packages"
PORTAGE_TMPDIR="/var/tmp"
USE="3dnow 3dnowex 3dnowext X Xaw3d aalib acl alsa apache2 apm avi bash-completion berkdb browserplugin bzip2 caps cdr cli cracklib crypt css curl cxx dbus dga dir
ectfb doc dri dts dvd encode erandom esd fbcon ffmpeg flac fltk force-cgi-redirect fortran gd gdbm ggi gif gpm gtk iconv imagemagick imlib innodb ipv6 jadetex java
 jpeg jpeg2k lcms lesstif libcaca libwww lirc lua mad mailwrapper matroska mbox mcal memlimit mikmod mmx mmx2 mmxext mng modules motif mozilla mpeg multislot mysql
 ncurses nls nptl ogg oggvorbis old-linux opengl openmp oss pam pcre pdflib perl png postscript python qt qt3support quicktime readline real samba sdl seccomp sess
ion slang snmp spell sqlite sse sse2 ssl svga tcpd tetex theora tiff truetype unicode usb userlocales v4l v4l2 vhosts videos vorbis webp wmf x264 x86 xattr xgetdef
ault xml xorg xosd xv xvid zlib" ABI_X86="32" ALSA_CARDS="ali5451 als4000 atiixp atiixp-modem bt87x ca0106 cmipci emu10k1 emu10k1x ens1370 ens1371 es1938 es1968 fm
801 hda-intel intel8x0 intel8x0m maestro3 trident usb-audio via82xx via82xx-modem ymfpci" APACHE2_MODULES="access auth auth_dbm auth_anon auth_basic auth_digest au
thz_host authz_default authz_user alias file-cache echo charset-lite cache disk-cache mem-cache ext-filter case_filter case-filter-in deflate mime-magic cern-meta 
expires headers usertrack unique-id proxy proxy-connect proxy-ftp proxy-http info include cgi cgid dav dav-fs vhost-alias speling rewrite log_config logio env sete
nvif mime status autoindex asis negotiation dir imap actions userdir so mem_cache mime_magic vhost_alias filter authn_file" APACHE2_MPMS="prefork" CALLIGRA_FEATURE
S="kexi words flow plan sheets stage tables krita karbon braindump author" COLLECTD_PLUGINS="df interface irq load memory rrdtool swap syslog" CPU_FLAGS_X86="3dnow
 3dnowext mmx mmxext popcnt sse sse2 sse3 sse4a" ELIBC="glibc" GPSD_PROTOCOLS="ashtech aivdm earthmate evermore fv18 garmin garmintxt gpsclock isync itrax mtk3301 
nmea ntrip navcom oceanserver oldstyle oncore rtcm104v2 rtcm104v3 sirf skytraq superstar2 timing tsip tripmate tnt ublox ubx" INPUT_DEVICES="evdev keyboard mouse" 
KERNEL="linux" LCD_DEVICES="bayrad cfontz cfontz633 glk hd44780 lb216 lcdm001 mtxorb ncurses text" LIBREOFFICE_EXTENSIONS="presenter-console presenter-minimizer" O
FFICE_IMPLEMENTATION="libreoffice" PHP_TARGETS="php5-6" PYTHON_SINGLE_TARGET="python2_7" PYTHON_TARGETS="python2_7 python3_4" QEMU_SOFTMMU_TARGETS="arm i386 x86_64
" QEMU_USER_TARGETS="alpha arm i386 x86_64" RUBY_TARGETS="ruby19 ruby20 ruby21 ruby23" USERLAND="GNU" VIDEO_CARDS="r128 radeon radeonhd vesa r300 r600" XTABLES_ADD
ONS="quota2 psd pknock lscan length2 ipv4options ipset ipp2p iface geoip fuzzy condition tee tarpit sysrq steal rawnat logmark ipmark dhcpmac delude chaos account"
USE_PYTHON="2.7 3.4"
Unset:  CC, CPPFLAGS, CTARGET, CXX, EMERGE_DEFAULT_OPTS, INSTALL_MASK, LANG, LC_ALL, PORTAGE_BUNZIP2_COMMAND, PORTAGE_COMPRESS, PORTAGE_COMPRESS_FLAGS, PORTAGE_RSY
NC_EXTRA_OPTS
```

**Comment by hkmaly** [comment]
Yeah. With moving /usr/lib/palemoon out of way, it installed.

**Comment by Massimo-B** [comment]
Sorry, but how did you get that far? www-client/palemoon-27.3.0 is failing here with
```
 0:31.10 js/src> /bin/sh: ./config.status: /mnt/ssd/tmp.notmpfs/portage/www-client/palemoon-27.3.0/work/palemoon-27.3.0/obj-x86_64-unknown-linux-gnu/_virtualenv/bin/pyt: bad interpreter: No such file or directory
```
Is that worth a bug report? I just thought the source package is not ready anyway with current gcc, so I just ignored and continued with the binary one.
Current gcc here is x86_64-pc-linux-gnu-6.3.0.

**Comment by hkmaly** [comment]
Not sure what your error message is, but I'm staying away from those bleeding edge gcc versions and use stable i686-pc-linux-gnu-4.9.3 ... oh, yes, also 32bit. I also don't have bleeding edge python. Both of those information are mentioned in the emerge --info I posted.

**Comment by deu** [comment]
This looks solved to me. Nothing can be done about it if it was caused by files from an ancient Pale Moon installation still in the system.

@Massimo-B's issue was probably fixed by b9a70ce158944f63a6ddf520ebedf60b65384c7f, so closing this.
